src/deploy.ts

```diff
@@ -0,0 +1,64 @@
+/*************************************************************************************************
+ *  Copyright (c) Red Hat, Inc. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE file in the project root for license information.
+ *************************************************************************************************/
+
+import * as ghCore from "@actions/core";
+import { Inputs } from "./generated/inputs-outputs";
+import { getSelector } from "./utils";
+import Oc from "./oc";
+
+namespace Deploy {
+    export async function newApp(): Promise<void> {
+        const image = ghCore.getInput(Inputs.IMAGE);
```

Comment:
We shouldn't be fetching the inputs here. The inputs should be fetched in the entrypoint run function (in index.ts) and passed around to each of the functions.

Reply:
Each input should only be fetched once

Reply:
Okay, much better